pulser-core/pulser/register/weight_maps.py

```diff
@@ -79,11 +79,12 @@ def get_qubit_weight_map(
         coords_arr = self.sorted_coords
         weights_arr = self.sorted_weights
         for qid, pos in qubits.items():
-            dists = np.round(
-                np.linalg.norm(coords_arr - np.array(pos), axis=1),
-                decimals=COORD_PRECISION,
+            matches = np.argwhere(
+                np.all(
+                    np.isclose(coords_arr, pos, atol=10 ** (-COORD_PRECISION)),
+                    axis=1,
+                )
```

Comment:
Actually, `np.isclose()` is better because now we are actually comparing each coordinate value with the relevant precision, while before we were (incorrectly) comparing the norm of the difference rounded to the precision.

The thing about what you suggest is that, in fact, it is possible that there is no match. Consider the case where the user does not define the detuning map from the register and then tries to see what the weights on each qubit would be; every qubit that does not align with a trap in the detuning map should have a weight of zero (ie it won't match with any trap).

Reply:
You are right, thanks for clarifying.
Then it's good to me, let's merge 😀